halo2_proofs/src/plonk.rs

```diff
@@ -159,6 +229,44 @@ impl<C: CurveAffine> ProvingKey<C> {
     pub fn get_vk(&self) -> &VerifyingKey<C> {
         &self.vk
     }
+    /// Writes a proving key to a buffer.
+    /// Does so by first writing the verifying key and then serializing the rest of the data (in the form of field polynomials)
+    pub fn write<W: io::Write>(&self, writer: &mut W) -> io::Result<()> {
+        self.vk.write(writer)?;
+        let partial_pkey = ProvingKeyWithoutVerifyingKey {
+            l0: self.l0.clone(),
+            l_last: self.l_last.clone(),
+            l_active_row: self.l_active_row.clone(),
+            fixed_values: self.fixed_values.clone(),
+            fixed_polys: self.fixed_polys.clone(),
+            fixed_cosets: self.fixed_cosets.clone(),
+            permutation: self.permutation.clone(),
+            ev: self.ev.clone(),
+        };
+        bincode::serialize_into(writer, &partial_pkey).expect("should be able to serialize pkey");
```

Comment:
I think we can add another variant in `plonk::Error` for pk/vk serde error (even just `Serde(err)` should be better than panic).